spring-cloud-aws-sns/src/main/java/io/awspring/cloud/sns/core/NotificationMessagingTemplate.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.awspring.cloud.sns.core;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import software.amazon.awssdk.services.sns.SnsClient;
+
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessagingException;
+import org.springframework.messaging.converter.CompositeMessageConverter;
+import org.springframework.messaging.converter.MessageConverter;
+import org.springframework.messaging.converter.StringMessageConverter;
+import org.springframework.messaging.core.AbstractMessageSendingTemplate;
+import org.springframework.messaging.core.DestinationResolver;
+import org.springframework.messaging.core.DestinationResolvingMessageSendingOperations;
+import org.springframework.messaging.core.MessagePostProcessor;
+
+import static io.awspring.cloud.sns.core.MessageHeaderCodes.NOTIFICATION_SUBJECT_HEADER;
+
+/**
+ * @author Alain Sahli
+ * @author Matej Nedic
+ * @since 1.0
+ */
+public class NotificationMessagingTemplate extends AbstractMessageSendingTemplate<TopicMessageChannel>
+		implements DestinationResolvingMessageSendingOperations<TopicMessageChannel> {
+
+	private final SnsClient snsClient;
+
+	private final DestinationResolver<String> destinationResolver;
+
+	public NotificationMessagingTemplate(SnsClient snsClient) {
+		this(snsClient, null);
+	}
+
+	public NotificationMessagingTemplate(SnsClient snsClient, MessageConverter messageConverter) {
```

Comment:
Should we take into account this in auto-configuration? How easy it is right now to send an object with assumption that its serialised to JSON?